source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,23 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
```

Comment:
You should use `std::string::back()` - O(1), `std::string::length()` - O(1) and `std::string::compare()` - O(n).

Previously, you were using `strlen()` - O(n) and `std::string::compare()` - O(n), i.e. walking the same string twice, which resulted in suboptimal performance, although the difference is probably negligible.